### PR TITLE
REST API_PURGE UE CONTEXT

### PR DESCRIPTION
**Pull request by aditya-mehta14**
This PR implements a REST API for purging the UE context. If the UE is in registered state, Network Initiated deregistration is triggered. Else, UE is directly removed. PFA for logs.
[amf.txt](https://github.com/omec-project/amf/files/8915724/amf.txt)
[gnbsim.txt](https://github.com/omec-project/amf/files/8915728/gnbsim.txt)

### CODE REVIEW

File: producer/oam.go [hunk 1]
```diff
@@ -39,6 +42,25 @@ type UEContext struct {
 
 type UEContexts []UEContext
 
+func HandleOAMPurgeUEContextRequest(supi, reqUri string, msg interface{}) (interface{}, string, interface{}, interface{}) {
+	amfSelf := context.AMF_Self()
+	if ue, ok := amfSelf.AmfUeFindBySupi(supi); ok {
```
Comment (vthiruveedula) [inline producer/oam.go:47]: can we send an error response (ex:404 not found) if supi doesn't exist in AMF? 
please add log if supi is not exist

Comment (aditya-mehta14) [reply producer/oam.go:47]: Changes done.

File: oam/api_purge_ue_context.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+	"github.com/omec-project/amf/context"
+	"github.com/omec-project/amf/logger"
+	"github.com/omec-project/amf/producer"
+	"github.com/omec-project/http_wrapper"
+
+	"github.com/omec-project/openapi/models"
+)
+
+func HTTPPurgeUEContext(c *gin.Context) {
```
[the remark this answers is not in the document]
Comment (aditya-mehta14) [reply oam/api_purge_ue_context.go:16]: Added a UT code for  HandleOAMPurgeUEContext method. Please check.

**Review by vthiruveedula — APPROVED** [review]
approving it as changes are as is

**Comment by badhrinathpa** [comment]
retest this please

**Comment by vthiruveedula** [comment]
retest this please

**Review by vthiruveedula — APPROVED** [review]

**Pull request merged**